enterprise/server/remote_execution/blockio/blockio.go

```diff
@@ -0,0 +1,100 @@
+package blockio
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"syscall"
+
+	"github.com/buildbuddy-io/buildbuddy/server/util/status"
+)
+
+// Store models a block-level storage system, which is useful as a backend for a
+// Network Block Device (NBD) or userfaultfd(2).
+type Store interface {
```

Comment:
Can you add some tests for this new interface and its implementations?

Reply:
Done